dockerfiles/Dockerfile

```diff
@@ -76,6 +76,66 @@ WORKDIR /srv/docsrs
 # Tini is a small init binary to properly handle signals
 CMD ["/usr/bin/tini", "/usr/local/bin/cratesfyi", "start-web-server", "0.0.0.0:80"]
 
+########################
+#  Build server stage  #
+########################
+
+FROM ubuntu:22.04 AS build-server
+
+RUN apt-get update \
+    && DEBIAN_FRONTEND=noninteractive apt-get upgrade -y \
+    && DEBIAN_FRONTEND=noninteractive apt-get install -y \
+        ca-certificates \
+        tini \
+        docker.io \
+        build-essential \
+        gcc \
+        pkg-config \
+    && rm -rf /var/lib/apt/lists/*
+
+COPY --from=build /build/target/release/cratesfyi /usr/local/bin
+
+WORKDIR /srv/docsrs
+# Tini is a small init binary to properly handle signals
+CMD ["/usr/bin/tini", "/usr/local/bin/cratesfyi", "start-build-server"]
+
+############################
+#  Registry watcher stage  #
+############################
+
+FROM ubuntu:22.04 AS registry-watcher
+
+RUN apt-get update \
+    && DEBIAN_FRONTEND=noninteractive apt-get upgrade -y \
+    && DEBIAN_FRONTEND=noninteractive apt-get install -y \
+        ca-certificates \
+        tini \
+        git \
+    && rm -rf /var/lib/apt/lists/*
+
+COPY --from=build /build/target/release/cratesfyi /usr/local/bin
+
+WORKDIR /srv/docsrs
+# Tini is a small init binary to properly handle signals
+CMD ["/usr/bin/tini", "/usr/local/bin/cratesfyi", "start-registry-watcher"]
```

Comment:
to replicated the planned production setup, the arguments should include: 
- `repository-stats-updater=enabled` (default: `disabled`) 
- for clarity: `cdn-invalidator=enabled` (default: `enabled`)

Reply:
except when infra would like to run these separately, I remember a discussion with pietry where he preferred all these three tasks in one process.